[quant][graphmode][fx][api] Call fuse in prepare

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#43984 [quant][graphmode][fx][api] Call fuse in prepare**
* #43983 [quant][graphmode][fx] Support inplace option
* #44042 [quant][graphmode][fx] Support quantize per channel in all cases

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

jerryzh168 committed:
[quant][graphmode][fx][api] Call fuse in prepare

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]

vkuzo commented on torch/quantization/quantize_fx.py:
```diff
@@ -21,10 +21,10 @@ def fuse_fx(graph_module, inplace=False):
     quantizer = Quantizer()
     prepare = quantizer.prepare_dynamic if is_dynamic_quant else quantizer.prepare
-    prepared = prepare(graph_module, qconfig_dict, inplace)
+    prepared = prepare(graph_module, qconfig_dict, inplace=True)
```
should this keep the value of inplace?

jerryzh168 replied:
inplace is applied in fuse_fx step already, e.g. when inplace is False, the model is already copied, we don't need to copy again

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][api] Call fuse in prepare"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459261](https://our.internmc.facebook.com/intern/diff/D23459261)

[ghstack-poisoned]
